### PR TITLE
SENT protocol capture implementation

### PR DESCRIPTION
**Pull request by dron0gus**

### CODE REVIEW

File: firmware/check.c [hunk 1]
```diff
@@ -14,11 +14,4 @@
 	#error "GCC compiler >= 9.3.1 required"
 #endif
 
-// Firmware builds but does not actually run on GCC10 :(
-// todo: one day someone should fix it, do YOU want to help?
-#if (GCC_VERSION > 1000)
```
Comment (rusefillc) [inline firmware/check.c:19]: why here? let's not?

Comment (dron0gus) [reply firmware/check.c:19]: I will cleanup this when it ready.

File: firmware/hw_layer/drivers/sent/sent.cpp [hunk 2]
```diff
@@ -13,89 +13,112 @@
 
 #include "sent.h"
 
-struct sent_channel {
+typedef enum
```
Comment (mck1117) [inline firmware/hw_layer/drivers/sent/sent.cpp:16]: if you really want to like c++, go for the magic of `enum class`!

Comment (dron0gus) [reply firmware/hw_layer/drivers/sent/sent.cpp:50]: Step by step. I got too many errors when try to convert to enum class.

**Comment by rusefillc** [comment]
We also have https://github.com/rusefi/rusefi/pull/4648

**Pull request merged**